backend/src/main/java/com/bakdata/conquery/sql/conversion/cqelement/ConceptPreprocessingService.java

```diff
@@ -0,0 +1,137 @@
+
+package com.bakdata.conquery.sql.conversion.cqelement;
+
+import java.time.LocalDate;
+import java.util.List;
+import java.util.Optional;
+
+import com.bakdata.conquery.apiv1.query.concept.filter.CQTable;
+import com.bakdata.conquery.apiv1.query.concept.filter.FilterValue;
+import com.bakdata.conquery.apiv1.query.concept.specific.CQConcept;
+import com.bakdata.conquery.models.common.Range;
+import com.bakdata.conquery.models.datasets.Column;
+import com.bakdata.conquery.sql.conversion.context.ConversionContext;
+import com.bakdata.conquery.sql.conversion.context.selects.ConceptSelects;
+import com.bakdata.conquery.sql.conversion.context.step.QueryStep;
+import com.bakdata.conquery.sql.conversion.dialect.SqlFunctionProvider;
+import org.jooq.Field;
+import org.jooq.impl.DSL;
+
+public class ConceptPreprocessingService {
+
+	private static final String DATE_RESTRICTION_COLUMN_NAME = "date_restriction";
+	private static final String VALIDITY_DATE_COLUMN_NAME = "validity_date";
+	private final CQConcept concept;
+	private final ConversionContext context;
+	private final SqlFunctionProvider sqlFunctionProvider;
+
+	public ConceptPreprocessingService(CQConcept concept, ConversionContext context) {
+		this.concept = concept;
+		this.context = context;
+		this.sqlFunctionProvider = this.context.getSqlDialect().getFunction();
+	}
+
+	/**
+	 * selects:
+	 * - (primary column)
+	 * - date restriction
+	 * - validity date
+	 * - any filter (group/event)
+	 * - any select (group/event)
+	 */
+	public QueryStep buildPreprocessingQueryStepForTable(String conceptLabel, CQTable table) {
+
+		ConceptSelects.ConceptSelectsBuilder selectsBuilder = ConceptSelects.builder();
+
+		selectsBuilder.primaryColumn(DSL.field(context.getConfig().getPrimaryColumn()));
+		selectsBuilder.dateRestriction(this.getDateRestrictionSelect(table));
+		selectsBuilder.validityDate(this.getValidityDateSelect(table));
+
+		List<Field<Object>> conceptSelectFields = this.getColumnSelectReferences(table);
+		List<Field<Object>> conceptFilterFields = this.getColumnFilterReferences(table);
+
+		// deduplicate because a concepts selects and filters can require the same columns
```

Comment:
sind die Jooq Fields nicht ausreichend equals/hashcode-d für ein Set?

Reply:
Uns war es im Endeffekt da zu unsicher, sich darauf zu verlassen, da in dem Interface keine Garantien gegeben werden